### ISSUE TITLE
minikube generates files in /tmp

### ISSUE THREAD

**Issue by bigon** [kubernetes/minikube#19515]
### What Happened?

Hello,

minikube seems to generate logs file in /tmp for each command that are run

For example, I run `minikube version` and I ends up with a file in `/tmp`: `/tmp/minikube_version_486aa15011e42ace60f28d9e74c31051228d0192_0.log`

Same with `minikube completion bash` for example. This is a problem since the command is run every-time I'm opening a shell

### Attach the log file

[log.txt](https://github.com/user-attachments/files/16750635/log.txt)

### Operating System

Other

### Driver

N/A

**Comment by afbjorklund** [comment]
You don't have to run the `completion` command every time, only when the minikube version changes...

Mostly it is saved to a file during packaging, for instance how it is done in the brew formula for minikube?

**Comment by bigon** [comment]
> You don't have to run the completion command every time, only when the minikube version changes...

Well the `completion` command prints some (ba)sh code that needs to be evaluated by the shell isn't it?

**Comment by afbjorklund** [comment]
Right, but you can save the output to a file

**Comment by Ritikaa96** [comment]
tmp files are supposed to be deleted after the user logs off , is it creating any issue for you though?

**Comment by Ritikaa96** [comment]
I found this issue: https://github.com/kubernetes/minikube/issues/2918
there are different cases discussed where tmp files are created. PTAL if you think your issue is covered by this one?

**Comment by bigon** [comment]
> tmp files are supposed to be deleted after the user logs off , is it creating any issue for you though?

Files in `/tmp` are deleted when the machines *reboots*, not when the user logoff

It's usually a bad idea to keep hundred of unneeded files around

All the other commands of the k8s ecosystem are not generating or deleting these files (ie `kubectl completion bash` is not generating these files)

Generate the files, do the needed stuff with it, delete the file is the usual pattern

**Comment by Ritikaa96** [comment]
> > tmp files are supposed to be deleted after the user logs off , is it creating any issue for you though?
> 
> Files in `/tmp` are deleted when the machines _reboots_, not when the user logoff
> 
> It's usually a bad idea to keep hundred of unneeded files around
> 
> All the other commands of the k8s ecosystem are not generating or deleting these files (ie `kubectl completion bash` is not generating these files)
> 
> Generate the files, do the needed stuff with it, delete the file is the usual pattern

Agreed , it is actual stress over the system in such case. Did you get the chance to look at the https://github.com/kubernetes/minikube/issues/2918 issue. the other scenarios are also discussed there

**Comment by bigon** [comment]
It's probably the same bug yes, but the other one is closed

**Comment by bigon** [comment]
/reopen

Still happening in 1.36.0
